cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowHttpHandlerTest.java

```diff
@@ -512,32 +509,37 @@ public void testWorkflowForkApp() throws Exception {
     waitState(programId, ProgramRunStatus.RUNNING.name());
 
     // Store the new RunRecord for the currently running run
-    historyRuns = getProgramRuns(programId, "running");
-    Assert.assertEquals(1, historyRuns.size());
-    runId = historyRuns.get(0).getPid();
+    runId = getRunIdOfRunningProgram(programId);
 
     // Wait till first action in the Workflow starts executing
-    verifyFileExists(Lists.newArrayList(firstSimpleActionFile));
+    verifyFileExists(Lists.newArrayList(firstFile));
 
     verifyRunningProgramCount(programId, runId, 1);
 
     // Signal the first action to continue
-    Assert.assertTrue(firstSimpleActionDoneFile.createNewFile());
+    Assert.assertTrue(firstDoneFile.createNewFile());
 
     // Wait till fork in the Workflow starts executing
-    verifyFileExists(Lists.newArrayList(oneSimpleActionFile, anotherSimpleActionFile));
+    verifyFileExists(Lists.newArrayList(branch1File, branch2File));
 
     // Two actions should be running in Workflow as a part of the fork
     verifyRunningProgramCount(programId, runId, 2);
 
     // Signal the Workflow that execution can be continued
-    Assert.assertTrue(oneSimpleActionDoneFile.createNewFile());
-    Assert.assertTrue(anotherSimpleActionDoneFile.createNewFile());
+    Assert.assertTrue(branch1DoneFile.createNewFile());
+    Assert.assertTrue(branch2DoneFile.createNewFile());
 
     // Workflow should now have one completed run
     verifyProgramRuns(programId, "completed");
   }
 
+  private String getRunIdOfRunningProgram(Id.Program programId) throws Exception {
```

Comment:
perhaps, but this particular test did not need it. It was always requesting the one and only RUNNING run record, which is a very specific use-case. We can perhaps create a ``getProgramRunIds(Id.Program programId, String status)`` for use in other tests, in that case ``getRunIdOfRunningProgram`` will be a wrapper over that method. But perhaps out of scope of this PR.